### ISSUE TITLE
Fatal Exception: java.lang.NullPointerException: Listener must not be null

### ISSUE THREAD

**Issue by bobymicroby** [patloew/RxLocation#55]
I think you have introduced this bug in 1.0.3

When you unsubscribe from ```LocationUpdatesFlowableOnSubscribe``` before ```onGoogleApiClientReady```  is called, there is no locationListener instance initialized and we got a NPE

```java
@Override
    protected void onUnsubscribed(GoogleApiClient apiClient) {
        LocationServices.FusedLocationApi.removeLocationUpdates(apiClient,  *locationListener*);
        locationListener.onUnsubscribed();
        locationListener = null;
    }


```
Fatal Exception: java.lang.NullPointerException: Listener must not be null
       at com.google.android.gms.common.internal.zzbq.checkNotNull(Unknown Source)
       at com.google.android.gms.common.api.internal.zzcp.zzb(Unknown Source)
       at com.google.android.gms.internal.zzcdh.zza(Unknown Source)
       at com.google.android.gms.common.api.internal.zzm.zzb(Unknown Source)
       at com.google.android.gms.common.api.internal.zzao.zze(Unknown Source)
       at com.google.android.gms.common.api.internal.zzbl.zze(Unknown Source)
       at com.google.android.gms.common.api.internal.zzbd.zze(Unknown Source)
       at com.google.android.gms.internal.zzccy.removeLocationUpdates(Unknown Source)
       at com.patloew.rxlocation.LocationUpdatesFlowableOnSubscribe.onUnsubscribed(LocationUpdatesFlowableOnSubscribe.java:54)
       at com.patloew.rxlocation.RxLocationFlowableOnSubscribe.lambda$subscribe$0(RxLocationFlowableOnSubscribe.java:58)
       at com.patloew.rxlocation.RxLocationFlowableOnSubscribe$$Lambda$1.cancel(Unknown Source)
       at io.reactivex.internal.disposables.CancellableDisposable.dispose(CancellableDisposable.java:49)
```

**Comment by kurtisnelson** [comment]
+1, I am seeing this on my emulators missing play services now that I added a timeout.

**Comment by patloew** [comment]
Fixed in v1.0.5.
